src/Bicep.Core/Syntax/ProgramSyntax.cs

```diff
@@ -5,30 +5,33 @@
 using System.Linq;
 using Bicep.Core.Parser;
 using Bicep.Core.Diagnostics;
+using Bicep.Core.Navigation;
 
 namespace Bicep.Core.Syntax
 {
     public class ProgramSyntax : SyntaxBase
     {
-        public ProgramSyntax(IEnumerable<SyntaxBase> statements, Token endOfFile, IEnumerable<Diagnostic> lexerDiagnostics)
+        public ProgramSyntax(IEnumerable<SyntaxBase> children, Token endOfFile, IEnumerable<Diagnostic> lexerDiagnostics)
         {
-            this.Statements = statements.ToList().AsReadOnly();
+            this.Children = children.ToImmutableArray();
             this.EndOfFile = endOfFile;
             this.LexerDiagnostics = lexerDiagnostics.ToImmutableArray();
         }
 
-        public IReadOnlyList<SyntaxBase> Statements { get; }
-
+        public ImmutableArray<SyntaxBase> Children { get; }
+        
         public Token EndOfFile { get; }
 
         public ImmutableArray<Diagnostic> LexerDiagnostics { get; }
 
         public override void Accept(SyntaxVisitor visitor)
             => visitor.VisitProgramSyntax(this);
 
-        public override TextSpan Span
-            => Statements.Any() ? 
-                TextSpan.Between(Statements.First(), EndOfFile) :
-                TextSpan.Between(EndOfFile, EndOfFile);
+        public override TextSpan Span => this.Children.Any()
+            ? TextSpan.Between(this.Children.First(), this.EndOfFile)
+            : this.EndOfFile.Span;
+
+        // TODO: Should we have a DeclarationSyntax abstract class?
```

Comment:
If there are places where you need to read this result and then cast to `IDeclarationSyntax`, then I think I'm in favor of a base class rather than interface.

Reply:
Yeah, I think I will do it in another PR.